woocommerce/woocommerce-gutenberg-products-block
# Replace filters with an extendibility API to hook into Cart and Checkout blocks

Fixes #3735.

This PR refactors the filter added in #3716 so instead of using `@wordpress/hooks`, it uses our own API. This adds an abstraction layer that should make life easier for extensions and it allows us to more easily make internal changes without breaking 3rd party extensions.

There are a couple of PRs that will need to be updated if we go ahead with this approach: #3750 and #3763. cc @opr 

### How to test the changes in this Pull Request:

1. Check out this branch in WC Subscriptions: `fix/filters-api` (see associated PR: 3979-gh-woocommerce/woocommerce-subscriptions).
2. Add a subscription product to your cart + any other product which is not a subscription.
3. Go to the Cart or Checkout blocks and verify the sidebar `Total` label is `Total due today`.
4. Remove the subscription from your cart and verify the `Total due today` label changes to `Total`.
